snuba/utils/streams/kafka.py

```diff
@@ -46,34 +77,76 @@ class KafkaConsumer(Consumer[TopicPartition, int, bytes]):
 
     def __init__(self, configuration: Mapping[str, Any]) -> None:
         self.__consumer = ConfluentConsumer(configuration)
```

Comment:
I'm probably going to remove `auto.offset.reset` from this configuration and force it to `error` since it can cause the consumer offset to move non-monotonically (or just generally erratically) without warning. My plan is to replace it with `initial.offset.reset` similar to how it is implemented for the [`SynchronizedConsumer`](https://github.com/getsentry/sentry/blob/95767d455b8004ec4b4c5026d84b64b6348e6d37/src/sentry/eventstream/kafka/consumer.py#L130), which has been sort of implemented here anyway.